resources/web/wwi/Parser.js

```diff
@@ -469,21 +517,112 @@ export default class Parser {
     const translation = convertStringToVec3(getNodeAttribute(node, 'translation', '0 0 0'));
     const scale = convertStringToVec3(getNodeAttribute(node, 'scale', '1 1 1'));
     const rotation = convertStringToQuaternion(getNodeAttribute(node, 'rotation', '0 0 1 0'));
+    let name = getNodeAttribute(node, 'name', '');
 
     let newNode;
-    if (type === 'track') {
+    if (type === 'track' || node.tagName === 'Track') {
       const geometriesCount = parseInt(getNodeAttribute(node, 'geometriesCount', '10'));
       newNode = new WbTrack(id, translation, scale, rotation, geometriesCount);
-    } else if (type === 'trackwheel') {
+    } else if (type === 'trackwheel' || node.tagName === 'Trackwheel') {
       const radius = parseFloat(getNodeAttribute(node, 'radius', '0.1'));
       const inner = getNodeAttribute(node, 'inner', '0').toLowerCase() === '1';
 
       newNode = new WbTrackWheel(id, translation, scale, rotation, radius, inner);
 
       parentNode.wheelsList.push(newNode);
-    } else if (node.tagName === 'Robot' || node.tagName === 'Solid' || type === 'solid' || type === 'robot')
-      newNode = new WbSolid(id, translation, scale, rotation);
-    else {
+    } else if (node.tagName === 'Robot' || node.tagName === 'Solid' || type === 'solid' || type === 'robot') {
+      if (name === '') {
+        if (node.tagName === 'Robot' || type === 'robot')
+          name = 'robot';
+        else
+          name = 'solid';
+      }
+      newNode = new WbSolid(id, translation, scale, rotation, name);
+    } else if (node.tagName === 'Accelerometer') {
+      if (name === '')
+        name = 'accelerometer';
+      newNode = new WbAccelerometer(id, translation, scale, rotation, name);
```

Comment:
```suggestion
      if (name === '')
        name = 'accelerometer';
      newNode = new WbAccelerometer(id, translation, scale, rotation, name === '' ? 'accelerometer' : name);
```

Reply:
would save us the ifs and the brackets, same for the others

Reply:
also I'm not sure to understand why we need for the name to be set, we don't seem to do so for non-devices. I don't believe we use this to distinguish between them (we shouldn't), so I don't understand

Reply:
We need the name for the documentation viewer.